lib/fields.nix

```diff
@@ -0,0 +1,33 @@
+{ lib }:
+{
+  /**
+    Creates a Field attribute set. mkField accepts an attribute set with the following keys:
+
+     Example:
+       mkField { }  // => { _type = "field"; }
+       mkField { default = "foo"; } // => { _type = "field"; default = "foo"; }
+       mkField { optional = true; } // => { _type = "field"; optional = true; }
+  */
+  mkField =
+    {
+      # Default value used when no definition is given in the configuration.
+      default ? null,
+      # Textual representation of the default, for the manual.
+      defaultText ? null,
+      # Example value used in the manual.
+      example ? null,
+      # String describing the field.
+      description ? null,
+      # Related packages used in the manual (see `genRelatedPackages` in ../nixos/lib/make-fields-doc/default.nix).
+      relatedPackages ? null,
```

Comment:
thought:
These could be bundled into a single `meta` attribute, and that might further improve performance by virtue of not being turned into values during non-docs evaluation.
I don't know how significant that is, and it may make `mkField` annoying to use, typing more and being dissimilar from `mkOption`.
Also the performance improvement could be inverted when any of the attrs inside of `meta` were to somehow become part of the main evaluation later (unlikely?).